airbyte-cdk/python/airbyte_cdk/sources/file_based/stream/default_file_based_stream.py

```diff
@@ -40,9 +41,15 @@ class DefaultFileBasedStream(AbstractFileBasedStream, IncrementalMixin):
     DATE_TIME_FORMAT = "%Y-%m-%dT%H:%M:%S.%fZ"
     ab_last_mod_col = "_ab_source_file_last_modified"
     ab_file_name_col = "_ab_source_file_url"
+    modified = "modified"
+    source_file_url = "source_file_url"
     airbyte_columns = [ab_last_mod_col, ab_file_name_col]
+    file_transfer_flag = "use_file_transfer"
+    use_file_transfer = False
```

Comment:
Should we remove one of these to avoid chance of them conflicting?

Reply:
Yes!

Reply:
Oh, no, sorry, I answered too fast; no, we shouldn't remove this one; the source resolves this to let the stream if It should handle a file transfer rather than transfer records. The source is the one responsible for doing whatever is needed to determine this, in this case now check what the transfer_type is with delivery_method.delivery_type.

Reply:
To extend more:

- file_transfer_flag = "use_file_transfer" is the kwarg we receive that we will pop.
- use_file_transfer = False the actual boolean/flag in the stream to be updated depending on `kwargs.pop(self.file_transfer_flag, False)`

But maybe I will reword to avoid confusion/

Reply:
Ok, I updated the key have constant format and name it FILE_TRANSFER_KW to not mix with previous flag we were using for sources.